flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/TableConfigOptions.java

```diff
@@ -178,14 +178,18 @@ private TableConfigOptions() {}
                                 + "functions, or data types will be persisted into the plan during "
                                 + "compilation. A schema allows for detecting incompatible changes "
                                 + "in the catalog during a plan restore operation. However, all "
-                                + "other metadata will still be retrieved from the catalog.")),
+                                + "other metadata will still be retrieved from the catalog. "
+                                + "This option does not apply to anonymous tables, functions and data types, "
```

Comment:
Move this to the option itself (not per enum value)? And also include temporary objects for which this option has also no effect.